collect_app/src/main/java/org/odk/collect/android/widgets/SelectWidget.java

```diff
@@ -111,6 +114,14 @@ public void onCompletion(MediaPlayer mediaPlayer) {
         super.playAllPromptText();
     }
 
+    @Override
+    protected void saveState() {
+        super.saveState();
+        if (searchStr != null) {
+            state.putString(SEARCH_TEXT + getPrompt().getIndex(), searchStr.getText().toString());
```

Comment:
Example of the above comment: if someone forgets to call `super.saveState();` in another child class before a line like this we'll get an NPE.